firedrake/dmplexinc.pxi

```diff
@@ -63,7 +65,7 @@ cdef extern from "petscsf.h":
     int PetscSFReduceBegin(PETSc.PetscSF,MPI.MPI_Datatype,const void*, void*,MPI.MPI_Op)
     int PetscSFReduceEnd(PETSc.PetscSF,MPI.MPI_Datatype,const void*, void*,MPI.MPI_Op)
 
-cdef extern from "petscbt.h":
+cdef extern from "petscbt.h" nogil:
```

Comment:
Does `nogil` makes things faster, or it helps avoiding deadlocks? What is the effect?